### ISSUE TITLE
Implement training loop

### ISSUE THREAD

**Issue by Supergecki** [Supergecki/Scientific-Computing-Project-WiSe2025_26-CIFAR-10-cGAN#7]
In `training/train.py`, implement the training loop with the following parameters:

- *Loss function*: Binary cross-entropy (vanilla GAN loss)
- *Optimizer*: Adam with beta_1=0.5, beta_2=0.999
- *Learning rates*: 2 * 10^-4 for both networks
- *Training ratio*: 1:1 (1 batch discriminator, 1 batch generator)
- *Batch size*: 64 or 128 (depending on whether you have a GPU ready? I don't at least)
